core/src/test/scala/org/apache/spark/rdd/PairRDDFunctionsSuite.scala

```diff
@@ -864,6 +873,45 @@ class NewFakeFormatWithCallback() extends NewFakeFormat {
   }
 }
 
+class YetAnotherFakeCommitter extends NewOutputCommitter with Assertions {
+  def setupJob(j: NewJobContext): Unit = {
+    JobID.jobid = j.getJobID().getId
+  }
+
+  def needsTaskCommit(t: NewTaskAttempContext): Boolean = false
+
+  def setupTask(t: NewTaskAttempContext): Unit = {
+    val rddid = t.getTaskAttemptID().getJobID().getId
```

Comment:
You don't need to ensure the stageId is equal to `rddId`, just make sure it inline with the jobId is okay.

Reply:
Thank you very much for the review. I noticed that the timeout issue that happens during the tests is actually because of my change. I didn't see the issue locally at first, since apparently I was testing on top of 2.2 branch. It seems that the committer actually look for the stageId and if there is no stage with the provided Id it will deny the commit. Since the change here proposes Rdd.id committer will always deny the commit. I will try to see if I can find a fix for this by just a small change by the end of this week. Otherwise I will close this PR to provide a more sophisticated solution for the issue.